src/pages/studyView/addChartButton/AddChartButton.tsx

```diff
@@ -433,13 +480,31 @@ class AddChartTabs extends React.Component<IAddChartTabsProps, {}> {
         tabs = _.map(
             this.props.store.genericAssayProfileOptionsByType.result,
             (options, type) => {
+                const molecularProfileIdsInType = options.find(
+                    option =>
+                        option.value ===
+                        this.selectedGenericAssayProfileIdByType.get(type)
+                )!.profileIds;
+                const entities = _.reduce(
+                    molecularProfileIdsInType,
+                    (set, profileId) => {
+                        set = new Set([
```

Comment:
this seems odd to create a new Set every iteration.  you cannot add them to existing set?